jscomp/runtime/js_iterator.ml

```diff
@@ -0,0 +1,6 @@
+type 'a t
+type 'a value = { done_ : bool; value : 'a option }
```

Comment:
doesn't this need `@mel.as "done"`?

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Iteration_protocols#done

Reply:
also: `value` can be omitted, so this is likely a `Js.nullable`

Reply:
I made both `done` and `value` nullable, because according to MDN they are both optional

Reply:
Btw could you clarify why `Js.nullable` and not `option`? Isn't `option` working fine with optional values ?

Reply:
I was thinking that because keys must be missing from the object, they'd have to be nullable. But I think we can change them back to option, especially because you can produce a `null` from an iterator, which `Js.nullable` would consider `None`.